TeamCode/src/main/java/com/acmerobotics/relicrecovery/drive/MecanumDrive.java

```diff
@@ -0,0 +1,114 @@
+package com.acmerobotics.relicrecovery.drive;
+
+import com.acmerobotics.relicrecovery.localization.Pose2d;
+import com.acmerobotics.relicrecovery.localization.Vector2d;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+/**
+ * Created by kelly on 9/28/2017.
+ *
+ * Wheel layout (top view):
+ *
+ *        FRONT
+ * (1)\\---------//(4)
+ *      |       |
+ *      |       |
+ *      |       |
+ *      |       |
+ * (2)//---------\\(3)
+ */
+public class MecanumDrive {
+
+    /***
+     * K = (wheelbase width + wheelbase height) / 4
+     */
+    public static double K = (18 + 18) / 4;
+
+    private DcMotor[] motors;
+    private int[] offsets;
+
+    /**
+     * construct drive with default configuration names
+     * @param map hardware map
+     */
+    public MecanumDrive(HardwareMap map) {
+        this(map, new String[]{"frontLeft", "rearLeft", "rearRight", "frontRight"});
+    }
+
+    /**
+     * construct drive with configuration names other than the default
+     * @param map hardware map
+     * @param names names of the motors in the hardware mapping
+     */
+    public MecanumDrive(HardwareMap map, String[] names) {
+        if (names.length != 4) {
+            throw new IllegalArgumentException("must be four for motors");
+        }
+        for (int i = 0; i < 4; i ++) {
+            motors[i] = map.get(DcMotor.class, names[i]);
```

Comment:
Any reason for `map.get(DcMotor.class, names[i])` instead of `map.dcMotors.get(names[i])`?

Reply:
no, I changed it